themes/src/main/resources/theme/base/login/resources/js/menu-button-links.js

```diff
@@ -0,0 +1,319 @@
+    switch (key) {
+      case " ":
+      case "Enter":
+      case "ArrowDown":
+      case "Down":
+        this.openPopup();
+        this.setFocusToFirstMenuitem();
+        flag = true;
+        break;
+
+      case "Esc":
+      case "Escape":
+        this.closePopup();
+        this.buttonNode.focus();
+        flag = true;
+        break;
+
+      case "Up":
+      case "ArrowUp":
+        this.openPopup();
+        this.setFocusToLastMenuitem();
+        flag = true;
+        break;
+
+      default:
+        break;
+    }
+
+    if (flag) {
+      event.stopPropagation();
+      event.preventDefault();
+    }
+  };
+
+  onButtonClick(event) {
+    if (this.isOpen()) {
+      this.closePopup();
+      this.buttonNode.focus();
+    } else {
+      this.openPopup();
+      this.setFocusToFirstMenuitem();
+    }
+
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
+  onMenuitemKeydown(event) {
+    const tgt = event.currentTarget;
+    const key = event.key;
+    let flag = false;
+
+    const isPrintableCharacter = (str) => str.length === 1 && str.match(/\S/);
+
+    if (event.ctrlKey || event.altKey || event.metaKey) {
+      return;
+    }
+
+    if (event.shiftKey) {
+      if (isPrintableCharacter(key)) {
+        this.setFocusByFirstCharacter(tgt, key);
+        flag = true;
+      }
+
+      if (event.key === "Tab") {
+        this.buttonNode.focus();
+        this.closePopup();
+        flag = true;
+      }
+    } else {
+      switch (key) {
+        case " ":
+          window.location.href = tgt.href;
+          break;
+
+        case "Esc":
+        case "Escape":
+          this.closePopup();
+          this.buttonNode.focus();
+          flag = true;
+          break;
+
+        case "Up":
+        case "ArrowUp":
+          this.setFocusToPreviousMenuitem(tgt);
+          flag = true;
+          break;
+
+        case "ArrowDown":
+        case "Down":
+          this.setFocusToNextMenuitem(tgt);
+          flag = true;
+          break;
+
+        case "Home":
+        case "PageUp":
+          this.setFocusToFirstMenuitem();
+          flag = true;
+          break;
+
+        case "End":
+        case "PageDown":
+          this.setFocusToLastMenuitem();
+          flag = true;
+          break;
+
+        case "Tab":
+          this.closePopup();
+          break;
+
+        default:
+          if (isPrintableCharacter(key)) {
+            this.setFocusByFirstCharacter(tgt, key);
+            flag = true;
+          }
+          break;
+      }
+    }
+
+    if (flag) {
+      event.stopPropagation();
+      event.preventDefault();
+    }
+  }
+
+  onMenuitemMouseover(event) {
+    const tgt = event.currentTarget;
+    tgt.focus();
+  }
+
+  onBackgroundMousedown(event) {
+    if (!this.domNode.contains(event.target)) {
+      if (this.isOpen()) {
+        this.closePopup();
+        this.buttonNode.focus();
+      }
+    }
+  }
+}
+
+// Initialize menu buttons
+
+window.addEventListener("load", () => {
```

Comment:
There should be no need for this in a module script, as these are [deferred by default](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#other_differences_between_modules_and_standard_scripts).

Reply:
Good one. Fixed.